core/src/mindustry/input/InputHandler.java

```diff
@@ -2163,6 +2164,15 @@ public void tryDropItems(@Nullable Building build, float x, float y){
         }
     }
 
+    public boolean canDepositItem(@Nullable Building build){
+        //takes advantage of itemDepositCooldown being able to be negative, allows the cooldown to be different for each turret
+        float cooldownOverride = 0;
+        if(build.block instanceof Turret turret && state.rules.enableTurretDepositCooldown && turret.depositCooldown >= 0){
```

Comment:
This should be a building method overridden by `Turret`, not something you have a special `instanceof` check for.